src/main/java/se/iths/crimedatabase/controller/AddressController.java

```diff
@@ -19,36 +22,51 @@ public AddressController(AddressService addressService) {
 
     @PostMapping
     public ResponseEntity<Address> create(@RequestBody Address address) {
+        if (address.getStreetAddress().isEmpty() || address.getCity().isEmpty() || address.getZipCode().isEmpty())
+            throw new BadRequestException("Street address, city and zipcode cannot be empty");
+
         Address createdAddress = addressService.create(address);
         return new ResponseEntity<>(createdAddress, HttpStatus.CREATED);
     }
 
     @DeleteMapping("{id}")
     public ResponseEntity<Void> delete(@PathVariable Long id) {
+        if (addressService.findById(id).isEmpty())
+            throw new NotFoundException(notFoundMessage(id));
+
         addressService.delete(id);
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
     @GetMapping("{id}")
     public ResponseEntity<Optional<Address>> findById(@PathVariable Long id) {
+        if (addressService.findById(id).isEmpty())
+            throw new NotFoundException(notFoundMessage(id));
+
         Optional<Address> foundAddress = addressService.findById(id);
         return new ResponseEntity<>(foundAddress, HttpStatus.OK);
     }
 
     @GetMapping
     public ResponseEntity<Iterable<Address>> findAll() {
         Iterable<Address> addresses = addressService.findAll();
+        if (!addresses.iterator().hasNext())
```

Comment:
Either we have this error or just return an empty json? I guess the error is better for clarity?

Reply:
To me a empty json is more confusing but if you're in agreement that it's better I'm down to change

Reply:
Will the program shut down when an exception is thrown? If it is, and we can't continue using it, we might want a json return. But otherwise I agree that an error is better

Reply:
Exceptions will not make the application shut down!

Reply:
Perfect!